src/main/java/com/damnhandy/uri/template/Expression.java

```diff
@@ -82,6 +85,14 @@
      */
     private Pattern matchPattern;
 
+    private String groupName = uid();
+
+    private static Random RANDOM = new SecureRandom();
```

Comment:
A good point. My original thought was that the collision chance of 12 bytes was practically none, and a counter could max out on a long running server. 
Now that you bring it up however, I think the performance hit of random is greater than the practical likelihood of using over 8 bytes of expressions (and if it becomes an issue, later code could handle it by resetting the counter or creating new counters and appending them together as strings).